File: src/main/java/vergisst/minecraftmod/weaponthrow/handler/EntityRegistry.java
package vergisst.minecraftmod.weaponthrow.handler;

import net.fabricmc.fabric.api.object.builder.v1.entity.FabricEntityTypeBuilder;
import net.minecraft.entity.EntityDimensions;
import net.minecraft.entity.EntityType;
import net.minecraft.entity.SpawnGroup;
import net.minecraft.registry.Registries;
import net.minecraft.registry.Registry;
import net.minecraft.util.Identifier;
import vergisst.minecraftmod.weaponthrow.WeaponThrow;
import vergisst.minecraftmod.weaponthrow.entity.WeaponThrowEntity;

public class EntityRegistry {
    public static final EntityType<WeaponThrowEntity> WEAPONTHROW = Registry.register(
            Registries.ENTITY_TYPE,
            new Identifier(WeaponThrow.MOD_ID, "weaponthrow"),
            FabricEntityTypeBuilder.<WeaponThrowEntity>create(SpawnGroup.MISC, WeaponThrowEntity::new).trackRangeBlocks(4).trackedUpdateRate(20).dimensions(EntityDimensions.fixed(0.5F, 0.5F)).build()
    );
    public static void registerEntities() {

    }
}

File: src/main/kotlin/vergisst/minecraftmod/weaponthrowlite/handlers/EntityRegistry.kt
package vergisst.minecraftmod.weaponthrowlite.handlers

import net.fabricmc.fabric.api.`object`.builder.v1.entity.FabricEntityTypeBuilder
import net.minecraft.entity.EntityDimensions
import net.minecraft.entity.EntityType
import net.minecraft.entity.SpawnGroup
import net.minecraft.registry.Registries
import net.minecraft.registry.Registry
import net.minecraft.util.Identifier

import vergisst.minecraftmod.weaponthrowlite.Weaponthrowlite
import vergisst.minecraftmod.weaponthrowlite.entity.WeaponThrowEntity

object EntityRegistry {
    val WEAPONTHROW: EntityType<WeaponThrowEntity> = Registry.register(
        Registries.ENTITY_TYPE,
        Identifier.of(Weaponthrowlite.MODID, "weaponthrow"),
        FabricEntityTypeBuilder
            .create(
            SpawnGroup.MISC
            // the closure here is alternative of ::WeaponThrowEntity in java
        ) { world, type ->
            WeaponThrowEntity(world, type)
        }
            .trackRangeBlocks(4)
            .trackedUpdateRate(20)
            .dimensions(EntityDimensions.fixed(0.5f, 0.5f))
            .build()
    )

    fun registerEntities() {}
}